api/src/main/java/com/redhat/ipaas/api/v1/rest/DataManager.java

```diff
@@ -57,68 +59,78 @@
 import java.util.function.Function;
 
 @ApplicationScoped
-public class DataManager {
+public class DataManager implements DataAccessObjectRegistry {
 
-    private static Logger logger = LoggerFactory.getLogger(DataManager.class.getName());
+    private static final Logger LOGGER = LoggerFactory.getLogger(DataManager.class.getName());
 
-    private ObjectMapper mapper = ObjectMapperHolder.OBJECT_MAPPER;
+    private ObjectMapper mapper;
+    private CacheContainer caches;
 
     // Inject optional data file by field injection.
     @Inject
     @ConfigurationValue("deployment.file")
     private String dataFileName;
 
-    private Cache<String, Map<String, WithId>> cache;
+    private final List<DataAccessObject> dataAccessObjects = new ArrayList<>();
+    private final Map<Class, DataAccessObject> dataAccessObjectMapping = new HashMap<>();
 
     // This is needed by CDI to create a proxy, but the actual instance created will use the constructor below which
     // has properly configured constructor injection.
     public DataManager() {
     }
 
     // Constructor to help with testing.
-    public DataManager(String dataFileName, Cache<String, Map<String, WithId>> cache) {
-        this(cache);
+    public DataManager(CacheContainer caches, ObjectMapper mapper, List<DataAccessObject> dataAccessObjects, String dataFileName) {
+        this(caches, mapper, dataAccessObjects);
         this.dataFileName = dataFileName;
     }
 
     // Inject mandatory via constructor injection.
     @Inject
-    public DataManager(Cache<String, Map<String, WithId>> cache) {
-        this.cache = cache;
+    public DataManager(CacheContainer caches, ObjectMapper mapper, List<DataAccessObject> dataAccessObjects) {
+        this.mapper = mapper;
+        this.caches = caches;
+        if (dataAccessObjects != null) {
+            this.dataAccessObjects.addAll(dataAccessObjects);
+        }
     }
 
     @PostConstruct
     public void init() {
-        if (cache.isEmpty() && dataFileName != null) {
+        if (dataFileName != null) {
             ReadApiClientData reader = new ReadApiClientData();
             try {
                 List<ModelData> mdList = reader.readDataFromFile(dataFileName);
                 for (ModelData modelData : mdList) {
                     addToCache(modelData);
                 }
             } catch (Exception e) {
```

Comment:
Should we rethrow this error after logging so we fail fast if we don't read the data successfully?